packages/access-token-jwt/test/dpop-verifier.test.ts

```diff
@@ -0,0 +1,1108 @@
+import { verifyDPoP } from '../src/dpop-verifier';
+import {
+  InvalidRequestError,
+  InvalidProofError,
+  InvalidTokenError,
+} from 'oauth2-bearer';
+import {
+  SignJWT,
+  generateKeyPair,
+  exportJWK,
+  calculateJwkThumbprint,
+} from 'jose';
+import crypto from 'node:crypto';
+import sinon from 'sinon';
+
+describe('verifyDPoP (real proofs)', () => {
+  let jwt: string;
+  let jwk: any;
+  let proof: string;
+  let headers: Record<string, any>;
+  let accessToken: string;
+  let jkt: string;
+  let url: string;
+  let method: string;
+
+  beforeEach(async () => {
+    const { publicKey, privateKey } = await generateKeyPair('ES256');
+    jwk = await exportJWK(publicKey);
+    accessToken = 'access.token.jwt';
+    jkt = await calculateJwkThumbprint(jwk);
+    method = 'GET';
+    url = 'https://api.example.com/resource';
+
+    const ath = crypto
+      .createHash('sha256')
+      .update(accessToken)
+      .digest('base64url');
+
+    proof = await new SignJWT({
+      htm: method,
+      htu: url,
+      iat: Math.floor(Date.now() / 1000),
+      jti: crypto.randomUUID(),
+      ath,
+    })
+      .setProtectedHeader({
+        typ: 'dpop+jwt',
+        alg: 'ES256',
+        jwk,
+      })
+      .sign(privateKey);
+
+    headers = {
+      authorization: `DPoP ${accessToken}`,
+      dpop: proof,
+    };
+    jwt = accessToken;
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  const validOptions = () => ({
+    jwt,
+    accessTokenClaims: { cnf: { jkt } },
+    headers,
+    method,
+    url,
+    supportedAlgorithms: ['ES256'],
+    iatOffset: 300,
+    iatLeeway: 60,
+  });
+
+  it('passes for a valid DPoP proof', async () => {
+    await expect(verifyDPoP(validOptions())).resolves.not.toThrow();
+  });
+
+  it('throws if Authorization header is missing', async () => {
+    delete headers.authorization;
+    await expect(verifyDPoP(validOptions())).rejects.toThrow(
+      InvalidRequestError
```

Comment:
In almost all these test cases the error properties (e.g. error message) are not being asserted, so there really is no way to know if the `InvalidRequestError` was thrown as a result of the intended scenario, or because of something else.